### PR TITLE
chore: allow for running OSS pipeline from monitor-ci script

### LINKED ISSUE #1007: Migrate CI back to OSS
everytime a change is made in a branch within the ui, our e2e tests get run against idpe's master branch
everytime a change is made in a branch within idpe, those same tests get run against the ui's master branch

this relationship is super dope and pushes the potential system wide breakage back to the person who unknowingly introduces it. we should have it for OSS too.

we currently have monitor-ci running the e2e tests against oss on every branch, the work is to copy that infrastructure over to influxdb and make that repo run the e2e changes against the ui's master branch on every change over there

### PR DESCRIPTION
**Pull request by wbaker85**
Partially closes #1007

Related: [monitor-ci PR#228](https://github.com/influxdata/monitor-ci/pull/228)

Updates the `run-monitor-ci-tests` bash script to have a different behavior if it is executed with the intent of testing changes in the OSS backend from `influxdb` with the latest UI from the `ui` repository.

If the script is executed with the env var `OSS_SHA` set, the API call to start the pipeline in `monitor-ci` will be what is required by the config updates in [monitor-ci PR#228](https://github.com/influxdata/monitor-ci/pull/228). If `OSS_SHA` is not set, the script will execute the same as before. The diff is a little misleading - basically all that is changing in the script is some updates to the comments at the beginning, and a conditional wrapping the existing pipeline API call for running from the `ui` repo, and the new logic for running with a set OSS backend commit sha against the latest UI.

We'll need to merge [monitor-ci PR#228](https://github.com/influxdata/monitor-ci/pull/228) first so that it can accept and act on the new `OSS_SHA` parameter. After this script update is merged, we can then update the `influxdb` CI to pull the script and execute it to run the e2e tests.

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 60 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1335)

File: scripts/run-monitor-ci-tests.bash [hunk 1]
```diff
@@ -4,87 +4,109 @@ set -eu -o pipefail
...
-if [[ "${UI_BRANCH}" == "master" ]]; then
-	DEPLOY_PROD=false # TODO: change this to true when we're ready to depend on this script
+	# start the monitor-ci pipeline from the influxdb repo
+	printf "\nstarting monitor-ci pipeline targeting monitor-ci branch ${MONITOR_CI_BRANCH} using OSS SHA ${OSS_SHA}\n"
+	pipeline=$(curl -s --fail --request POST \
+		--url https://circleci.com/api/v2/project/gh/influxdata/monitor-ci/pipeline \
+		--header "Circle-Token: ${API_KEY}" \
+		--header 'content-type: application/json' \
+		--header 'Accept: application/json'    \
+		--data "{\"branch\":\"${MONITOR_CI_BRANCH}\", \"parameters\":{ \"oss-sha\":\"${OSS_SHA}\" }}")
```
Comment (genehynson) [inline scripts/run-monitor-ci-tests.bash:108]: could we simplify this if/else by having one curl request outside the if/else and just pass different objects to --data? Then the else block would just be creating the data object for OSS

File: scripts/run-monitor-ci-tests.bash [hunk 2]
```diff
@@ -4,87 +4,109 @@ set -eu -o pipefail
 
 ########################
 # --- Script Summary ---
```
Comment (genehynson) [inline scripts/run-monitor-ci-tests.bash:6]: thanks for updating this comment!

File: scripts/run-monitor-ci-tests.bash (hunk 1 above)
Comment (wbaker85) [reply scripts/run-monitor-ci-tests.bash]: Good idea! I pushed a change for this, it does simplify things a bit.

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?vars%5Blighthouse_score_branch%5D=1335)

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 56 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](true1335)

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 55 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](true1335)

**Review by genehynson — APPROVED** [review]
Looks good!

**Pull request merged**